src/scripts/button.js

```diff
@@ -0,0 +1,42 @@
+import { html, render } from 'lit-html';
+
+class Button extends HTMLElement {
+  constructor() {
+    super();
+
+    this.attachShadow({ mode: 'open' });
+
+    this.htmlElement = (label, color) => html`
+      <style>
+        button {
+          cursor: pointer;
+          color: #fff;
+          border: 0;
+          padding: 8px 10px;
+          border-radius: 4px;
+        }
+
+        .green {
+          background-color: #28e236;
+        }
+
+        .red {
+          background-color: #e22828;
+        }
+      </style>
+      <button class="${color}" @click=${this.handleClick}>${label}</button>
+    `;
+  }
+
+  handleClick = () => {
+    this.onClick();
+    render(this.htmlElement(this.label, this.color), this.shadowRoot);
```

Comment:
You could have used render lifecycle method from lit-element instead on doing lit-html way